### DISCUSSION TITLE
Is there any documentation on the template system?

### DISCUSSION THREAD

**Discussion by BloodyRain2k** [rr-/szurubooru#670] [Q&A]
I've been making some small modifications to the client now and I'm sometimes left scratching my head on how the templates work.

So far I've been able to find everything I need, one way or another, from looking through the existing templates, but the understanding I'm getting from that is limited.

For example: `<%= ctx.makeFlexboxAlign() %>` seems to "return" the result into the HTML.
Yet here `<input type='text' id='merge-id-<%-ctx.name %>' pattern='^[0-9]+$' value='<%- ctx.post ? ctx.post.id : '' %>'/>` the `<%-` seems to also "return" the result into the HTML...

Now I'm confused why there's two ways of doing that.

Something that'd be really interesting would be debugging, because often I manage to break everything in a way that I get a 0 byte app.min.js and NO error message as to WHAT I broke...

**Comment by neobooru** [comment]
I can't fully answer your question, as I don't know exactly how everything works.
We use https://underscorejs.org/#template for the template stuff. And I would recommend to take a dive into the build.js script because that might have some answers.
I've wanted to use vue for the client for maybe 5 years now, but it's just a lot of work. Besides, I am not a frontend dev so I wouldn't even know the best way to structure things. Tbh I think I started a private rewrite 3 times already, but each time I find a edge case which is hard to 'update to modern standards'. E.g. stuff like the 'ios-inner-height' package can probably be replaced by modern css. A lot of our styling is also outdated I guess. Flexbox can solve a lot of complexity. And the build system can also be modernized by vite, webpack, {insert other build tool of the year}.
Oh and the same can be said for the server, though in a lesser degree. Nowadays you'd probably want to use a well known framework like flask, fastapi, whatever.
E: Oh and you can try building the client with `node .\build.js --watch --no-transpile` (not sure if --no-transpile is needed) and inspecting the created app.min.js file. My debugging mostly consists of well placed console.log calls though.
